packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useEditorSearchExtension.ts

```diff
@@ -49,99 +37,7 @@ const searchOverlay = (query: RegExp) => {
 	} };
 };
 
-const addCodeMirrorExtension = (CodeMirror: CodeMirror5Emulation) => {
-	CodeMirror.defineOption('joplin.search-highlight-state', null, ()=>{});
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
-	CodeMirror?.defineExtension('setMarkers', function(keywords: any, options: any) {
```

Comment:
This pull request converts `setMarkers` from a CodeMirror 5 legacy extension to a regular function. One of the issues that b3bd4e03b902911a0b3c84c5e9f4cfc07cb40988 tried to address was that legacy CodeMirror 5 extensions are defined for all editor instances. As a result, `clearMarkers`/`setMarkers` would often use state associated with the wrong editor. This issue is easier to to address if the `setMarkers` function is a standard function that accepts an editor as input.